CrossPare/src/main/java/de/ugoe/cs/cpdp/eval/AbstractWekaEvaluation.java

```diff
@@ -507,4 +538,79 @@ public static double getNECM(Evaluation eval, double factor) {
         return (eval.numFalsePositives(1)+factor*eval.numFalseNegatives(1))/
                 (eval.numInstances());
     }
+
+    /**
+     * <p>
+     * Calculates the bias of a data set. That means the percentage of instances in
+     * the training data that are defective.
+     * </p>
+     *
+     * @param data data set to calculate the bias for
+     * @return bias
+     */
+    public static double getBias(Instances data) {
+        double count = 0.0;
+        for (Instance instance : data) {
+            if (instance.classValue() > 0.0) {
+                count += 1.0;
+            }
+        }
+        return (count / data.size());
+    }
+
+    /**
+     * <p>
+     * Calculates ratio of the effort of the largest percentile of defect artifacts by
+     * the overall amount of defect artifacts.
+     * </p>
+     *
+     * @param data test data set
+     * @param efforts effort list from the test data set
+     * @return Effort of the largest percentile of defect artifacts by overall amount of defect artifacts.
+     */
+    public static double getProp1Defects(Instances data, List<Double> efforts) {
+        List<Double> effortDefect = new ArrayList<>();
+        int amountDefectArtifacts = 0;
+        for (int i = 0; i < data.size(); i++){
+            if (data.get(i).classValue() > 0.0) {
+                amountDefectArtifacts += 1;
+                effortDefect.add(efforts.get(i));
+            }
+        }
+        effortDefect.sort(Collections.reverseOrder());
+        double percentile = Math.ceil(effortDefect.size() / 100.0);

```

Comment:
should use int as datatype, since you are actually calculating an index here. maybe also rename to percentile_index?